### PR TITLE
Add and use TypeAlias & TypeUnion to handle aliases with type-params.

### PR DESCRIPTION
**Pull request by epabst**
Also:
Move null/lambda checks out of toKotlinParam.
Combine ...IfStandardType methods and fix method names.

### CODE REVIEW

File: testData/typeAlias/unionTypeParams.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+type Values<V> = List<V>;
+type ListOrMultiMap<K,V> = List<V> | Map<K,Values<V>>;
```
Comment (epabst) [inline testData/typeAlias/unionTypeParams.d.ts]: It seems difficult to find the right balance between using Kotlin's alias type and dealing with TS' union type.  TS aliases can contain unions.  I'd prefer to defer that for now for that reason.

File: testData/typeAlias/unionTypeParams.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+declare var fooMap: ListOrMultiMap<string,number>;
+
+declare function mapKey(a: ListOrMultiMap<number,String>);
+
+declare var fooStringOrMap: ListOrMultiMap<string,number>;
```
Comment (epabst) [inline testData/typeAlias/unionTypeParams.d.ts]: Deuped.

**Comment by epabst** [comment]
@bashor: Updated.

**Comment by bashor** [comment]
Thanks

**Comment by bashor** [comment]
Thanks

File: testData/typeAlias/unionTypeParams.d.ts (hunk 1 above)
Comment (epabst) [inline testData/typeAlias/unionTypeParams.d.ts]: Also, Kotlin's alias type isn't fully released yet.  I'm ok with using an early access for ts2kt, but not for the Kotlin it generates until that is released.

Comment (bashor) [inline testData/typeAlias/unionTypeParams.d.ts]: Another (very important) reason is that we can expand functions with union type in the signature (generate overloads).

**Pull request merged**